tests/test-data/deterministic/models/infer.wppl

```diff
@@ -0,0 +1,9 @@
+var model = function() {
+  return flip(1);
+};
+[
+  // Test the use of ES6 object literal shorthand.
+  sample(Infer({model})),
+  sample(Infer({model: model})),
+  sample(Infer({model() { return model(); }}))
```

Comment:
Exactly. Note that esprima does distinguish between the long and short literal syntax with an extra flag on the property node, but this doesn't make it through our transforms, which is why code gen always outputs the long hand version.

I added these tests to make sure we find out if something occurs which breaks this. At some point I imagine switching all of the examples and inference tests over to the short hand syntax, but I figured I'd wait to see whether it actually becomes idiomatic first. (Though I'm increasingly confident that it will, so I'd be happy to change it soon if there are no objections.)